prow/jobs.md

```diff
@@ -70,7 +70,90 @@ that matches `trigger` will suffice. This is useful if you want to make one
 command that reruns all jobs. If unspecified, the default configuration makes
 `/test <job-name>` trigger the job.
 
-### Pod Utilities
+## Standard Triggering and Execution Behavior for Jobs
+
+When configuring jobs, it is necessary to keep in mind the set of rules Prow has
+for triggering jobs, the GitHub status contexts that those jobs provide, and the
+rules for protecting those contexts on branches.
+
+### Triggering Jobs
+#### Trigger Types
+
+`prow` will consider three different types of jobs that run on pull requests
+(presubmits):
+
+ 1. jobs that run unconditionally and automatically. All jobs that set
+     `always_run: true` fall into this set.
+ 2. jobs that run conditionally, but automatically. All jobs that set
+    `run_if_changed` to some value fall into this set. 
+ 3. jobs that run conditionally, but not automatically. All jobs that set
+    `run_always: false` and do not set `run_if_changed` to any value fall
+    into this set and require a human to trigger them with a command.
+
+By default, jobs fall into the third category and must have their `always_run` or
+`run_if_changed` configured to operate differently.
+
+In the rest of this document, "a job running unconditionally" indicates that the
+job will run even if it is normally conditional and the conditions are not met.
+Similarly, "a job running conditionally" indicates that the job runs if all of its
+conditions are met.
+
+#### Triggering Jobs With Comments
+
+A developer may trigger jobs by posting a comment to a pull request that contains
+one or more of the following phrases:
+ - `/test job-name` : When posting `/test job-name`, any jobs with matching triggers
+   will be triggered unconditionally.
+ - `/retest` : When posting `/retest`, two types of jobs will be triggered:
+   - all jobs that have run and failed will run unconditionally
+   - any not-yet-executed automatically run jobs will run conditionally 
+ - `/test all` : When posting `/test all`, all automatically run jobs will run
+   conditionally.
+   
+Note: is is possible to configure a job's `trigger` to match any of the above keywords
```

Comment:
Why don't we just drop support for customizing how to trigger jobs? It is archaic functionality.

Reply:
Out of scope, but kind of agreed -- we name our jobs to be unique `pull-ci-org-repo-branch-name` but want `/test name`